movebucket_test.go

```diff
@@ -0,0 +1,185 @@
+package bbolt_test
+
+import (
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	bolt "go.etcd.io/bbolt"
+	"go.etcd.io/bbolt/internal/btesting"
+)
+
+func TestBucket_MoveBucket(t *testing.T) {
+	testCases := []struct {
+		name          string
+		srcBucketPath []string
+		dstBucketPath []string
+		isInlined     bool
+		expErr        error
+	}{
+		{
+			name:          "happy path",
+			srcBucketPath: []string{"x", "y", "z"},
+			dstBucketPath: []string{"a", "b"},
+			isInlined:     false,
+			expErr:        nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(*testing.T) {
+			db := btesting.MustCreateDB(t)
+
+			// arrange
+			if err := db.Update(func(tx *bolt.Tx) error {
+				bk := createBucketIfNotExist(t, tx, tc.srcBucketPath...)
+				insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
+				createBucketIfNotExist(t, tx, tc.dstBucketPath...)
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			// act
+			//var tmpFile string
+			if err := db.Update(func(tx *bolt.Tx) error {
+				srcBucket := retrieveParentBucket(t, tx, tc.srcBucketPath...)
+				dstBucket := retrieveChildBucket(t, tx, tc.dstBucketPath...)
+				bucketToMove := tc.srcBucketPath[len(tc.srcBucketPath)-1]
+
+				//// dump bucketToMove to bbolt file for assertion
```

Comment:
@ahrtr  I am trying to dump the bucket content into file before moving for assertion after moving

i could not write it to boltDB file, because this requires a separate db to be opened in order to match the bbolt file specifications. 

please correct me if i am wrong